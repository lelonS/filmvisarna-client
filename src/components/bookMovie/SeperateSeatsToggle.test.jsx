// Seperate is not a word, it should be separate

import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import SeperateSeatsToggle from './SeperateSeatsToggle';


// Mocks
const mockUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockUsedNavigate,
}));

const mockObject = { toggle: undefined };
jest.mock('react-easier', () => ({
  ...jest.requireActual('react-easier'),
  useStates: () => mockObject
}));

window.scrollTo = jest.fn();

describe('SeperateSeatsToggle', () => {

  test('renders correctly', () => {
    render(<SeperateSeatsToggle />);
    expect(screen.getByText('VÃ¤lj separata platser')).toBeInTheDocument();
    expect(screen.getByRole('checkbox')).toBeInTheDocument();

  });

  test('renders correctly with snapshot', () => {
    const tree = renderer.create(<SeperateSeatsToggle />).toJSON();
    expect(tree).toMatchSnapshot();

  });

  test('updates input values correctly', () => {
    const setSeats = jest.fn();
    render(<SeperateSeatsToggle setSeats={setSeats} />);

    const checkbox = screen.getByRole('checkbox');

    // Simulate changing checkbox using fireEvent
    fireEvent.click(checkbox);
    expect(mockObject.toggle).toBe(true);

    expect(setSeats).toHaveBeenCalledTimes(1);
    expect(setSeats).toHaveBeenCalledWith([]);
  });

});