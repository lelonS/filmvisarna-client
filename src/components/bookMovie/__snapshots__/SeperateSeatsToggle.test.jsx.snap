// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeperateSeatsToggle renders correctly with snapshot 1`] = `
<label
  className="flex items-center cursor-pointer"
  htmlFor="toogleA"
>
  <div
    className="relative"
  >
    <input
      className="sr-only"
      id="toogleA"
      onChange={[Function]}
      type="checkbox"
    />
    <div
      className="w-10 h-4 bg-gray-400 rounded-full shadow-inner"
    />
    <div
      className="dot absolute w-6 h-6 bg-white rounded-full shadow -left-1 -top-1 transition"
    />
  </div>
  <div
    className="ml-3 text-white font-medium"
  >
    VÃ¤lj separata platser
  </div>
</label>
`;
